### PR TITLE
Mingw enhancement

### PR DESCRIPTION
**Pull request by scienceasdf**
Support for MinGW python extension build and matlab mex file build.

### CODE REVIEW

**Review by bcoconni — CHANGES_REQUESTED** [review]
Thanks for the PR.

Some modifications to your code are needed since you are requesting manual modifications to Python/Anaconda files which is not our way to proceed. Instead this should be handled automatically by overloading the `distutils` compiler code.

Also we should trigger the CI compilation of Python modules for MinGW (which basically consists in reversing the commit 10530b9905738fec00d6b97b0de42167716b550f) to check that the build of the Python modules is successful.

**Comment by bcoconni** [comment]
FYI we previously investigated how to build Python modules with MinGW (see issue #285) but came to the conclusion that this was not feasible since Python officially dropped the support of MinGW. I'd be interested to have your feedback about the discussion we had in issue #285, and I'd be very glad if we could work around these limitations with your PR.

File: python/fpectl/setup.py.in [hunk 1]
```diff
@@ -56,6 +56,9 @@ compiler = new_compiler(compiler=compiler_name)
 if compiler.compiler_type == 'unix':
     cpp_compile_flags = ['-std=c++11']
     cpp_link_flags = []
+elif compiler.compiler_type == 'mingw32':
+    cpp_compile_flags = ['-std=c++11', '-D_hypot=hypot']
```
[the remark this answers is not in the document]
Comment (scienceasdf) [reply python/fpectl/setup.py.in:60]: Thanks for your review. 

I use this macro as `Python.h`  header file declares `::hypot` function. I'm using anaconda with python 3.6, whose distutils may contain some modification code. 

I close this PR for the moment. My branch will be working in progress for better MinGW compability support, to pass the CI python module building. If any new findings, I will be back here.:grin:

[the remark this answers is not in the document]
Comment (bcoconni) [reply python/fpectl/setup.py.in:60]: Ha OK, I did not think to check `Python.h`. That's a good point :smile:

[the remark this answers is not in the document]
Comment (bcoconni) [reply python/fpectl/setup.py.in:60]: And we'll be more than happy to provide you some help so do not hesitate to re-open this PR if you need some support.

**Pull request closed without merging**